### PR TITLE
Implement new sponsorship system

### PR DESCRIPTION
**Pull request by ewdurbin**
This work sets out to unify PSF Foundation, PyCon, and PyPI sponsorship applications, benefits, and management "under one roof" at python.org!

### CODE REVIEW

File: sponsors/forms.py [hunk 1]
```diff
@@ -2,43 +2,48 @@
...
+        return packages_benefits
 
     def clean(self):
         cleaned_data = super().clean()
 
         if not any([cleaned_data.get(bp.name) for bp in self.benefits_programs]):
-            raise forms.ValidationError(_("You have to pick a minimum number of benefits."))
+            raise forms.ValidationError(
+                _("You have to pick a minimum number of benefits.")
+            )
```
Comment (berinhard) [inline sponsors/forms.py:69]: @ewdurbin are you using black to format the code?

Comment (ewdurbin) [reply sponsors/forms.py:69]: Yes, but it is not enforced in any way.

Comment (ewdurbin) [reply sponsors/forms.py:69]: I've generally been applying black formatting to each application that gets "major" work. So for this work I've been running `black sponsors/` before committing.

Comment (berinhard) [reply sponsors/forms.py:69]: I'm a +1 for using black. I'll run it before my next commits

**Comment by berinhard** [comment]
@bwaliszewski commits b331015 and 5a9e49a are my final commits to this PR. Let me know what do you think about it and if there's something we still have to change to unblock the merge.

**Comment by berinhard** [comment]
@ewdurbin commits b331015 and 5a9e49a are my final commits to this PR. Let me know what do you think about it and if there's something we still have to change to unblock the merge.

**Pull request merged**